eth_defi/gmx/plans.md

```diff
@@ -0,0 +1,429 @@
+# GMX Integration for eth_defi: Implementation Plan
+
+## Goal
+
+The goal of this project is to integrate GMX with eth_defi, enabling users to interact with GMX's decentralized perpetual futures exchange through eth_defi's infrastructure. This integration will allow users to execute trades, manage positions, and access real-time and historical market data directly from eth_defi.
+
+## Use Cases
+
+1. **User Trading**: User X wants to open a leveraged long position on WETH using USDC as collateral. For this, we create a function `open_position` that interacts with GMX's smart contracts to execute the trade.
+2. **Position Management**: User Y wants to close a portion of their existing short position on ETH and withdraw part of their collateral. For this, we create a function `manage_position` that handles partial closures and collateral withdrawals.
```

Comment:
Two separate concerns and need their own functions, through can be combined
- Opening and closing positions: also should be called increasing and decreasing position, because sometimes you want to modify existing position - whether GMX can do this, I do not know
- Managing collateral

Because what you can do is that you deposit GMX once, then keep trading with the same collateral.